### PR TITLE
feat: create example folder with example plugin

### PR DESCRIPTION
**Pull request by monilpat**
# Relates to:
Initial setup of example plugin structure, to come examples of adapters, clients, services, etc. This will make it easier to standardize development, best practices, code review, and ultimately quality of the repo.

# Risks
Low - This is an example plugin that serves as a template/reference implementation

# Background

## What does this PR do?
Creates an example plugin folder with a sample implementation including:
- Basic plugin structure
- Sample actions, providers, and evaluators
- TypeScript configuration
- Build setup with tsup
- NPM package configuration

## What kind of change is this?
Features (non-breaking change which adds functionality)

# Documentation changes needed?
My changes do require a change to the project documentation which is included. Will update the global readme once all examples are added.

# Testing

## Where should a reviewer start?
1. Review the plugin structure in `packages/_examples/plugin/`
2. Key files to review:
   - `src/plugins/samplePlugin.ts` - Main plugin definition
   - `src/actions/sampleAction.ts` - Example action implementation
   - `src/providers/sampleProvider.ts` - Example provider implementation
   - `src/evaluators/sampleEvaluator.ts` - Example evaluator implementation

## Detailed testing steps
1. Navigate to the plugin directory:

### CODE REVIEW

File: packages/_examples/plugin/package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (cygaar) [inline packages/_examples/plugin/package.json:1]: You'll want to exclude this from the lerna config so we don't accidentally publish this plugin (unless for some reason you want to)

File: packages/_examples/plugin/src/plugins/samplePlugin.ts
Comment (odilitime) [inline packages/_examples/plugin/src/plugins/samplePlugin.ts:1]: why not just index.ts why a subfolder called plugins? Are you hoping a plugin can include more than one plugin?

File: packages/_examples/plugin/package.json (hunk 1 above)
Comment (monilpat) [reply packages/_examples/plugin/package.json:1]: Yeah that is a good call out will do :)

File: packages/_examples/plugin/src/plugins/samplePlugin.ts
Comment (monilpat) [reply packages/_examples/plugin/src/plugins/samplePlugin.ts:1]: We introduced the plugins subfolder structure to accommodate scenarios where a single "umbrella" plugin might contain multiple, more specialized plugins or modules. For example, Coinbase could have distinct functionalities like trading, token transfers, and commerce invoice creation. Organizing these capabilities within a dedicated directory allows for better modularity, scalability, and maintainability as the plugin’s functionality grows over time.

**Pull request merged**